src/net_processing.cpp

```diff
@@ -3072,6 +3053,26 @@ bool PeerLogicValidation::ProcessMessages(CNode* pfrom, std::atomic<bool>& inter
     return fMoreWork;
 }
 
+void PeerLogicValidation::ProcessBlockValidationResponse(CNode* pfrom, const std::shared_ptr<const CBlock> pblock, const CBlockIndex* pindex, const BlockValidationResponse& validation_response)
+{
+    LOCK(cs_main);
+
+    // If we've reconstructed this block via compactblocks then
+    // Clear download state for this block, which is in
+    // process from some other peer.  We do this after calling
+    // ProcessNewBlock so that a malleated cmpctblock announcement
+    // can't be used to interfere with block relay.
```

Comment:
I'm out of my depth here - should be looked at by someone with lots of net_processing experience.